### ISSUE TITLE
The plugin "theme" created a page with a component that doesn't exist (`path.resolve` doesn't work while `require.resolve` works)

### ISSUE THREAD

**Issue by deadcoder0904** [gatsbyjs/gatsby#12447]
## Description

I am using Gatsby Themes & my folder structure is like 👇

```
packages/
   theme/
      src/
        blog/
        template/
           blog-post.js
site/
    src/
       blog/
```

And in `gatsby-node.js`, I have `createPages` lifecycle method which takes in `component`.

The `component` works only if I use `require.resolve(`./src/templates/blog-post.js`)`

And does not work if I use `path.resolve(`./src/templates/blog-post.js`)`

This is because I don't have `./src/templates/blog-post.js` in `site` directory but I do have it in `theme` directory.

I think this should be mentioned in the docs since the error was confusing me for a while but I watched the live stream by Jason about Gatsby Themes & remembered to replace `path` with `require` & it worked.

For context, the error was `The plugin "theme" created a page with a component that doesn't exist`. Maybe provide a `path` to the component so it's easier to figure it out :)

**Comment by sidharthachatterjee** [comment]
Thank you for opening this, @deadcoder0904 

We'd love to accept an addition to the docs!

cc @ChristopherBiscardi

**Comment by ChristopherBiscardi** [comment]
Absolutely, the different ways of handling paths are critical to understanding how to author themes. This is one of the aspects of themes that is more about helping people understand the publishing and writing of npm packages that are going to be in the theme docs.

I submitted a PR to handle adding the component path to the error message. https://github.com/gatsbyjs/gatsby/pull/12472

**Comment by sidharthachatterjee** [comment]
Published in gatsby@2.1.28
